src/lib/root_prover/root_prover.ml

```diff
@@ -160,27 +166,26 @@ module Make (Inputs : Inputs_intf) :
         ( Merkle_list.verify ~init:root_hash merkle_list best_tip_hash
         |> Option.is_some )
     in
-    let root_with_validation =
-      External_transition.skip_time_received_validation
-        `This_transition_was_not_received_via_gossip
-        (External_transition.Validation.wrap root_transition_with_hash)
-    in
-    let best_tip_with_validation =
-      External_transition.skip_time_received_validation
-        `This_transition_was_not_received_via_gossip
-        (External_transition.Validation.wrap best_tip_with_hash)
-    in
-    let%bind validated_root =
-      Deferred.map
-        (External_transition.validate_proof ~verifier root_with_validation)
-        ~f:(Result.map_error ~f:(Fn.const (Error.of_string "invalid proof")))
-    in
-    let%map validated_best_tip =
-      Deferred.map
-        (External_transition.validate_proof ~verifier best_tip_with_validation)
-        ~f:(Result.map_error ~f:(Fn.const (Error.of_string "invalid proof")))
-    in
-    (validated_root, validated_best_tip)
+    let open Deferred.Result.Monad_infix in
+    External_transition.(
+      Deferred.Result.combine
+        ( Validation.wrap root_transition_with_hash
```

Comment:
The two statements in combine look redundant. Can you make them into a function?